spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java

```diff
@@ -524,6 +524,7 @@ private void bootstrapRetryTopicIfNecessary() {
 		BeanDefinitionRegistry registry = (BeanDefinitionRegistry) this.beanFactory;
 		if (!registry.containsBeanDefinition(RetryTopicInternalBeanNames
 				.RETRY_TOPIC_BOOTSTRAPPER)) {
+			this.logger.warn(() -> "Could not bootstrap RetryTopic early. Bootstrapping lazily.");
```

Comment:
Why do we need this warn?
Looks like this one is going to happen in 99% application in the field.

What is wrong with just silently registering that bean if it does not present yet?
We do something like this conditional registration all the time and no warnings.

Reply:
So, this is why I got the impression that you didn't get the context and wrote the comment below. Actually it's the opposite - I expect all of the applications to bootstrap the feature via the `RegistryPostProcessor`, and only if for some unexpected reason the `RPP`  is not triggered properly this logic should be executed.

Since that's an unexpected path, I'd rather know that this is happening so we can properly fix it, and when we get enough confidence we should be able to remove this logic.